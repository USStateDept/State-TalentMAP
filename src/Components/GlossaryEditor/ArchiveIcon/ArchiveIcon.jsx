import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import InteractiveElement from '../../InteractiveElement';

class ArchiveIcon extends Component {
  constructor(props) {
    super(props);
    this.submitOption = this.submitOption.bind(this);
    this.state = {
      isArchived: this.props.isArchived,
    };
  }

  componentWillReceiveProps(nextProp) {
    // Reset the isArchived state to the object's value if
    // hasErrored becomes true after the initial mount.
    if (nextProp.hasErrored) {
      this.setState({ isArchived: nextProp.isArchived });
    }
  }

  submitOption() {
    const { id, onSubmitOption } = this.props;
    this.setState({ isArchived: !this.state.isArchived });
    onSubmitOption({ id, is_archived: !this.state.isArchived });
  }

  render() {
    const { isArchived } = this.state;
    // set initial values as if term is NOT archived
    let icon = 'eye-slash';
    let className = 'term-not-archived';
    let text = 'Archive this term';

    if (isArchived) {
      icon = 'eye';
      className = 'term-archived';
      text = 'Un-archive this term';
    }

    return (
      <InteractiveElement role="link" className={`archive-link ${className}`} onClick={this.submitOption}>
        <FontAwesome name={icon} />{text}
      </InteractiveElement>
    );
  }
}

ArchiveIcon.propTypes = {
  onSubmitOption: PropTypes.func.isRequired,
  isArchived: PropTypes.bool,
  id: PropTypes.number.isRequired,
};

ArchiveIcon.defaultProps = {
  isArchived: false,
};

export default ArchiveIcon;
