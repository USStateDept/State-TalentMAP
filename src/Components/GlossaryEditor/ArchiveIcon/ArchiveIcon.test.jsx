import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ArchiveIcon from './ArchiveIcon';

describe('ArchiveIconComponent', () => {
  const props = {
    onSubmitOption: () => {},
    isArchived: false,
    id: 1,
  };

  it('is defined', () => {
    const wrapper = shallow(<ArchiveIcon {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('can call the onSubmitOption function', () => {
    const spy = sinon.spy();
    const wrapper = shallow(<ArchiveIcon {...props} onSubmitOption={spy} />);
    wrapper.find('InteractiveElement').simulate('click');
    sinon.assert.calledOnce(spy);
  });

  it('sets the correct values when isArchived is true', () => {
    const wrapper = shallow(<ArchiveIcon {...props} isArchived />);
    expect(wrapper.find('InteractiveElement').render().text()).toBe('Un-archive this term');
    expect(wrapper.find('.term-archived').exists()).toBe(true);
  });

  it('sets the correct values when isArchived is false', () => {
    const wrapper = shallow(<ArchiveIcon {...props} isArchived={false} />);
    expect(wrapper.find('InteractiveElement').render().text()).toBe('Archive this term');
    expect(wrapper.find('.term-not-archived').exists()).toBe(true);
  });

  it('resets the isArchived state if an error is passed after initial mount', () => {
    const wrapper = shallow(<ArchiveIcon {...props} isArchived={false} />);
    expect(wrapper.instance().state.isArchived).toBe(false);
    wrapper.find('InteractiveElement').simulate('click');
    expect(wrapper.instance().state.isArchived).toBe(true);
    wrapper.setProps({ hasErrored: true });
    expect(wrapper.instance().state.isArchived).toBe(false);
  });

  it('matches snapshot when isArchived is false', () => {
    const wrapper = shallow(<ArchiveIcon {...props} isArchived={false} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when isArchived is true', () => {
    const wrapper = shallow(<ArchiveIcon {...props} isArchived />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
