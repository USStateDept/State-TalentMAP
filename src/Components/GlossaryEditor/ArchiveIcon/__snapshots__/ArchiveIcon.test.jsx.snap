// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArchiveIconComponent matches snapshot when isArchived is false 1`] = `
<InteractiveElement
  className="archive-link term-not-archived"
  onClick={[Function]}
  role="link"
  type="div"
>
  <FontAwesome
    name="eye-slash"
  />
  Archive this term
</InteractiveElement>
`;

exports[`ArchiveIconComponent matches snapshot when isArchived is true 1`] = `
<InteractiveElement
  className="archive-link term-archived"
  onClick={[Function]}
  role="link"
  type="div"
>
  <FontAwesome
    name="eye"
  />
  Un-archive this term
</InteractiveElement>
`;
